datasets/us/fincen_special_measures/crawler.py

```diff
@@ -0,0 +1,147 @@
+        pdf_link_rescinded = BASE_URL + pdf_link_rescinded
+
+    # Create and add details to the sanction
+    sanction = h.make_sanction(context, entity)
+    sanction.add("country", "us")
+    finding_date = row.get("finding", "")
+    nprm_date = row.get("notice_of_proposed_rulemaking", "")
+    listing_date = finding_date if finding_date else nprm_date
+    listing_date = convert_date(listing_date)
+    sanction.add("listingDate", listing_date)
+    final_rule_date = row.get("final_rule", "")
+    if final_rule_date != "---":
+        sanction.add("startDate", convert_date(final_rule_date))
+
+    rescinded_date = row.get("rescinded", "")
+    if rescinded_date != "---":
+        sanction.add("endDate", convert_date(rescinded_date))
+
+    # Add description with the PDF link
+    if pdf_link_finding:
+        sanction.add("description", pdf_link_finding)  # Finding PDF link
+    if pdf_link_nprm:
+        sanction.add("description", pdf_link_nprm)  # NPRM PDF link
+    if pdf_link_final_rule:
+        sanction.add("description", pdf_link_final_rule)  # Final Rule PDF link
+    if pdf_link_rescinded:
+        sanction.add("description", pdf_link_rescinded)  # Rescinded PDF link
+
+    # Emit the entity and the sanction
+    context.emit(entity, target=True)
+    context.emit(sanction)
+
+
+# Parse the table and yield rows as dictionaries.
+def parse_table(table: html.HtmlElement) -> Generator[Dict[str, str], None, None]:
+    headers = None
+    for row in table.findall(".//tr"):
+        if headers is None:
+            headers = [
+                (
+                    slugify(el.text_content(), sep="_")
+                    if el.text_content().strip()
+                    else "company"
+                )
+                for el in row.findall("./th")
+            ]
+            headers.extend(
+                [
+                    "pdf_link_finding",
+                    "pdf_link_nprm",
+                    "pdf_link_final_rule",
+                    "pdf_link_rescinded",
+                ]
+            )
+            continue
+        cells = [el.text_content().strip() for el in row.findall("./td")]
+
+        # Extract PDF links from the relevant columns
+        pdf_link_finding = None
+        pdf_link_nprm = None
+        pdf_link_final_rule = None
+        pdf_link_rescinded = None
+
+        pdf_links = row.findall(".//a[@href]")
+
+        # Assuming the order of links in the row corresponds to the columns: Finding, NPRM, and Final Rule.
+        for i, pdf_link_element in enumerate(pdf_links):
+            pdf_link = pdf_link_element.get("href")
+            if pdf_link.endswith(".pdf"):
+                if i == 0:
+                    pdf_link_finding = pdf_link
+                elif i == 1:
+                    pdf_link_nprm = pdf_link
+                elif i == 2:
+                    pdf_link_final_rule = pdf_link
+                elif i == 3:
+                    pdf_link_rescinded = pdf_link
```

Comment:
I think it's better to keep the table parsing simpler, and do cell.findall("a") in crawl_item instead. Then you don't have to worry about offsets (and sometimes there are more than one link in a cell), and you know exactly which column the link related to.

```python
def parse_table(table: HtmlElement) -> Generator[Dict[str, str], None, None]:
    headers = None
    for row in table.findall(".//tr"):
        if headers is None:
            headers = []
            for el in row.findall("./th"):
                # Workaround because lxml-stubs doesn't yet support HtmlElement
                # https://github.com/lxml/lxml-stubs/pull/71
                eltree = cast(HtmlElement, el)
                headers.append(slugify(eltree.text_content()))
            continue

        cells = row.findall("./td")  # <--- edited here
        assert len(headers) == len(cells), (headers, cells)
        yield {hdr: c for hdr, c in zip(headers, cells)}
```

and then `finding_date = row.get("finding").text_content()` and `pdf_link_finding = row.get("finding").find("./a/@href")` or something

The risk with assuming column order is that they might add/change columns and if you don't have an assertion on the column headings being in the order you assume, the code might continue emitting data but it becomes invalid.